examples/java/src/test/java/org/apache/beam/examples/cookbook/TriggerExampleTest.java

```diff
@@ -112,15 +122,26 @@ public void testTotalFlow () {
         .apply(Window.<KV<String, Integer>>into(FixedWindows.of(Duration.standardMinutes(1))))
         .apply(new TotalFlow("default"));
 
-    PCollection<TableRow> results =  totalFlow.apply(ParDo.of(new FormatResults()));
+    PCollection<String> results =  totalFlow.apply(ParDo.of(new FormatResults()));
 
-
-    PAssert.that(results).containsInAnyOrder(OUT_ROW_1, OUT_ROW_2);
+    PAssert.that(results)
+        .containsInAnyOrder(canonicalFormat(OUT_ROW_1), canonicalFormat(OUT_ROW_2));
     pipeline.run();
 
   }
 
-  static class FormatResults extends DoFn<TableRow, TableRow> {
+  // Sort the fields and toString() the values, since TableRow has a bit of a dynamically
+  // typed API and equals()/hashCode() are not appropriate for matching in tests
+  static String canonicalFormat(TableRow row) {
```

Comment:
Yea, that's a nice idea. Filed as a starter project [BEAM-332](https://issues.apache.org/jira/browse/BEAM-332). Today we don't expose `SerializableMatchers` so it would have only limited applicability until such time as we sort that out.